src/samples/PoseWithCovariance.hpp

```diff
@@ -0,0 +1,72 @@
+#ifndef BASE_SAMPLES_POSE_WITH_COVARIANCE_HPP
+#define BASE_SAMPLES_POSE_WITH_COVARIANCE_HPP
+
+#include <string>
+#include <base/Time.hpp>
+#include <base/TransformWithCovariance.hpp>
+#include <base/samples/RigidBodyState.hpp>
+
+namespace base { namespace samples {
+
+/**
+ * Represents a 3D pose with uncertainty of the frame 'object_frame_id' in the frame 'frame_id'.
+ * A frame ID shall be globally unique and correspond to a coordinate system.
+ * Following the [Rock's conventions](http://rock.opendfki.de/wiki/WikiStart/Standards)
+ * coordinate systems are right handed with X forward, Y left and Z up.
+ *
+ * The pose \f$ ^{reference}_{object}T \f$ in this structure transforms a vector \f$ v_{object} \f$
+ * from the object frame to the reference frame:
+ * \f$ v_{reference} = ^{reference}_{object}T \cdot v_{object} \f$
+ * Example of a transformation chain:
+ * \f$ ^{world}_{sensor}T = ^{world}_{body}T \cdot ^{body}_{sensor}T \f$
```

Comment:
My point was about the usability of such a trivial method with the necessity to throw, not it's implementation. 

Since you think it would be helpful I implemented it. I was about to implement a `inverse()` method as well, but since for TransformWithCovariance the result of  `tr * tr2.inverse()` is covariance-wise unequal to `tr.compositionInv(tr2)` I think it would be misleading or even dangerous. I'd rather implement `compositionInv` and `preCompositionInv` as well, but I don't see much benefit in forwarding all those methods.